openmls/src/treesync/diff.rs

```diff
@@ -290,31 +307,27 @@ impl<'a> TreeSyncDiff<'a> {
         sender_leaf_index: LeafIndex,
         key_package: KeyPackage,
         path: Vec<ParentNode>,
-    ) -> Result<(), TreeSyncDiffError> {
+    ) -> Result<(), ApplyUpdatePathError> {
         let parent_hash =
             self.process_update_path(backend, ciphersuite, sender_leaf_index, path)?;
 
         // Verify the parent hash.
         let phe = key_package
             .extension_with_type(ExtensionType::ParentHash)
-            .ok_or(TreeSyncDiffError::MissingParentHash)?;
+            .ok_or(ApplyUpdatePathError::MissingParentHash)?;
         let key_package_parent_hash = phe
             .as_parent_hash_extension()
             .map_err(|_| LibraryError::custom("no parent hash extension"))?
             .parent_hash();
         if key_package_parent_hash != parent_hash {
-            return Err(TreeSyncDiffError::ParentHashMismatch);
+            return Err(ApplyUpdatePathError::ParentHashMismatch);
         };
 
         // Replace the leaf.
-        let node = Node::LeafNode(LeafNode::new(key_package, backend.crypto()).map_err(|e| {
-            if let KeyPackageError::CryptoError(e) = e {
-                TreeSyncDiffError::CryptoError(e)
-            } else {
-                TreeSyncDiffError::LibraryError(LibraryError::custom("key package error"))
-            }
-        })?);
-        self.diff.replace_leaf(sender_leaf_index, node.into())?;
+        let node = Node::LeafNode(LeafNode::new(key_package, backend.crypto())?);
+        self.diff
+            .replace_leaf(sender_leaf_index, node.into())
+            .map_err(|_| LibraryError::custom("Expected sender leaf to be in the tree"))?;
```

Comment:
Could you add this as assumption on the input to the function docs? Just so the person working on #804 has it on their radar.

Reply:
I added TODO #804 in all these places

Reply:
👍🏻 thanks!